packages/volto/src/components/manage/AnchorPlugin/components/LinkButton/AddLinkForm.jsx

```diff
@@ -1,324 +1,446 @@
+  const userId = token ? jwtDecode(token).sub : '';
+
+  const Select = reactSelect.default;
+
+  // Effect to set the initial selected value when data.url changes
+  useEffect(() => {
+    if (data.url) {
+      const initialUrl = isInternalURL(data.url)
+        ? flattenToAppURL(data.url)
+        : data.url;
+
+      setSelectedValue({ label: initialUrl, value: initialUrl });
+      setSearchTerm(initialUrl);
+      setLink(initialUrl);
+    } else {
+      setSelectedValue(null);
+      setSearchTerm('');
+      setLink('');
+    }
+  }, [data.url]);
+
+  useEffect(() => {
+    const delayDebounceFn = setTimeout(() => {
+      const searchParams = searchTerm
+        ? { SearchableText: `${searchTerm}*` }
+        : {};
+
+      dispatch(
+        searchContent(
+          '/',
+          {
+            ...searchParams,
+            Creator: userId,
+            sort_on: 'modified',
+            sort_order: 'descending',
+            metadata_fields: '_all',
+            b_size: searchTerm ? 10 : 5,
+            ...(objectBrowserPickerType ? { portal_type: ['Image'] } : {}),
+          },
+          'latestContent',
+        ),
+      );
+
+      dispatch(
+        searchContent(
+          contextUrl,
+          {
+            ...searchParams,
+            'path.depth': 1,
+            sort_on: 'getObjPositionInParent',
+            metadata_fields: '_all',
+            b_size: searchTerm ? 10 : 5,
+            ...(objectBrowserPickerType ? { portal_type: ['Image'] } : {}),
+          },
+          'crtFolderContent',
+        ),
+      );
+
+      dispatch(
+        searchContent(
+          '/',
+          {
+            ...searchParams,
+            sort_on: 'getObjPositionInParent',
+            metadata_fields: '_all',
+            b_size: searchTerm ? 10 : 5,
+            ...(objectBrowserPickerType ? { portal_type: ['Image'] } : {}),
+          },
+          'allFoldersContent',
+        ),
+      );
+    }, 500);
+
+    return () => clearTimeout(delayDebounceFn);
+  }, [dispatch, contextUrl, userId, searchTerm, objectBrowserPickerType]);
+
+  const content = useMemo(() => {
+    const latestContent = subrequests?.['latestContent']?.items || [];
+    const crtFolderContent = subrequests?.['crtFolderContent']?.items || [];
+    const allFoldersContent = subrequests?.['allFoldersContent']?.items || [];
+
+    const contentOptions = [
+      ...latestContent,
+      ...crtFolderContent,
+      ...allFoldersContent,
+    ];
+
+    return uniqBy(contentOptions, '@id');
+  }, [subrequests]);
+
+  const filteredSuggestions = useMemo(() => {
+    const suggestions = content
+      .filter((item) => {
+        if (filterSuggestionByType) {
+          return item['@type'] === filterSuggestionByType;
+        }
+        return item;
+      })
+      .map((item) => ({
+        label: item.title,
+        value: item['@id'],
+        src: `${item['@id']}/@@images/image/preview`,
```

Comment:
The preview scale is 400x400, we should probably use a smaller one. Probably thumb, or icon.